src/Server.php

```diff
@@ -152,7 +152,26 @@ private function handle(string $method, string $path, $contents, $request) : Res
 				if ($filesystem->has($path) === true) {
 					$mimetype = $filesystem->getMimetype($path);
 					if ($mimetype === self::MIME_TYPE_DIRECTORY) {
+						$contentType= explode(";", $request->getHeaderLine("Content-Type"))[0];
+
 						$filename = $this->guid();
+						// FIXME: make this list complete for at least the things we'd expect (turtle, n3, jsonld, ntriples, rdf);
```

Comment:
This could be done by using code from `pdsinterop/flysystem-rdf`:

```php
$formats = new \Pdsinterop\Rdf\Formats(); 
$format = $formats->getFormatForMime($mime);
$extension = $formats->getExtensionsForFormat($format)[0];
```

It would be trivial to add an `getExtensionForMime` there to do this...